Added OTP to the magic links guide (#4235)
FusionAuth/fusionauth-site: astro/src/content/docs/lifecycle/authenticate-users/passwordless/magic-links.mdx

<YouTube id="hMqxo68ZJlw" />

## Magic Links And One-Time Passwords

This feature in FusionAuth sends a secure, single-use, time-bound, code. By default, if a user enters an email address, this is sent via email. If the user enters a phone number, this is sent via SMS. When a user visits the link or enters a one-time code, they are logged in, as if by magic. The one-time code is also known as a one-time password or OTP.

> The user don't need to provide any other credentials; ownership and access to the delivered code is considered proof of who they are. This is also known as "magic link" or OTP login. It does not require a password, hence the general term "passwordless".

Reviewer: ```suggestion
The user don't need to provide any other credentials; ownership and access to the delivered code is considered proof of identity. This is also known as "magic link" or OTP login. It does not require a password, hence the general term "passwordless".
```
very slight wording tweak
